coral/src/app/features/topics/overview/components/TopicOverviewResourceTabs.tsx

```diff
@@ -0,0 +1,96 @@
+import { Tabs } from "@aivenio/aquarium";
+import { NavigateFunction, Outlet, useNavigate } from "react-router-dom";
+import {
+  TopicOverviewTabEnum,
+  TOPIC_OVERVIEW_TAB_ID_INTO_PATH,
+  isTopicsOverviewTabEnum,
+} from "src/app/router_utils";
+import PreviewBanner from "src/app/components/PreviewBanner";
+
+type Props = {
+  currentTab: TopicOverviewTabEnum;
+  topicName: string;
+};
+
+function TopicOverviewResourcesTabs({ currentTab, topicName }: Props) {
+  const navigate = useNavigate();
+
+  function navigateToTab(
+    navigate: NavigateFunction,
+    resourceTypeId: unknown
+  ): void {
+    if (isTopicsOverviewTabEnum(resourceTypeId)) {
+      navigate(
+        `/topic/${topicName}/${TOPIC_OVERVIEW_TAB_ID_INTO_PATH[resourceTypeId]}`,
+        {
+          replace: true,
+        }
+      );
+    }
+  }
+
+  const tabsMap: Array<{
+    topicOverviewTabEnum: TopicOverviewTabEnum;
+    title: string;
+  }> = [
+    {
+      topicOverviewTabEnum: TopicOverviewTabEnum.OVERVIEW,
+      title: "Overview",
+    },
+    {
+      topicOverviewTabEnum: TopicOverviewTabEnum.ACLS,
+      title: "Subscriptions",
+    },
+    {
+      topicOverviewTabEnum: TopicOverviewTabEnum.MESSAGES,
+      title: "Messages",
+    },
+    {
+      topicOverviewTabEnum: TopicOverviewTabEnum.SCHEMA,
+      title: "Schema",
+    },
+    {
+      topicOverviewTabEnum: TopicOverviewTabEnum.DOCUMENTATION,
+      title: "Documentation",
+    },
+    {
+      topicOverviewTabEnum: TopicOverviewTabEnum.HISTORY,
+      title: "History",
+    },
+    {
+      topicOverviewTabEnum: TopicOverviewTabEnum.SETTINGS,
+      title: "Settings",
+    },
+  ];
+
+  return (
+    <div>
+      <Tabs
+        value={currentTab}
+        onChange={(resourceTypeId) => navigateToTab(navigate, resourceTypeId)}
+      >
+        {tabsMap.map((tab) => {
+          return (
+            <Tabs.Tab
+              title={tab.title}
+              value={tab.topicOverviewTabEnum}
+              aria-label={tab.title}
+              key={tab.title}
+            >
+              {currentTab === tab.topicOverviewTabEnum && (
+                <div>
+                  <PreviewBanner
+                    linkTarget={`/topicOverview?topicname=${topicName}`}
+                  />
+                  <Outlet />
```

Comment:
Should we pass the selected topic name and env in the context prop so they are accessible in the child components?

In otherwords:
<Outlet context={{ topicName, env }} />
... then in child component
const { topicName, env } = useOutletContext();

Ref: https://reactrouter.com/en/6.4.4/hooks/use-outlet-context

Reply:
This would be handy when making requests in the tabs, as they take these params.

Reply:
This might be pretty nifty instead of picking them from the URL using `useParams` and/or `useRouter` :o

Reply:
Yeah, with `useParams` i think we could cover topic name but for example the selected dev does not seem to be part of the url.

Reply:
Yes we could do that! 💡  

I'll add the environments later in a different PR (waiting for decision for the UI) but I can already pass the `topicName` through with this PR.

Reply:
Yeah the prop of having the env as a search param would be that users would be able to sent a link directly and if someone opens it, they have the environment already selected. But that's not a HUGE usability thing, especially since we don't know how often you would actually sent a direct link to anyone 🤔

Reply:
Added passing the topicName through context! 

I added a custom hook in the parent component - it's recommended in the docs to enable proper typing!

Added a `TopicOverview` component with dummy content and name as usecase.